libp2p/network/url_conversion.go

```diff
@@ -35,25 +41,38 @@ func Libp2pMultiaddrFromServiceURL(serviceURL string) (multiaddr.Multiaddr, erro
 		peerIPVersionStr = ipVersion4
 	)
 
-	peerUrl, err := url.Parse(anyScheme + serviceURL)
+	// Conditionally add the anyScheme prefix if no scheme is present.
+	if strings.Contains(serviceURL, "://") {
+		return nil, fmt.Errorf("usage of scheme is invalid service URL: %s", serviceURL)
+	} else {
+		serviceURL = anyScheme + serviceURL
+	}
+
+	peerUrl, err := url.Parse(serviceURL)
 	if err != nil {
-		return nil, fmt.Errorf(
-			"parsing peer service URL: %s: %w",
-			serviceURL,
-			err,
-		)
+		return nil, fmt.Errorf("parsing peer service URL: %s: %w", serviceURL, err)
+	}
+
+	// Check if service URL contains a port number
+	if _, port, err := net.SplitHostPort(peerUrl.Host); err != nil || port == "" {
+		return nil, fmt.Errorf("missing port number and delimiter in service URL: %s", serviceURL)
 	}
 
 	peerIP, err := getPeerIP(peerUrl.Hostname())
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("resolving peer IP for hostname: %s: %w", peerUrl.Hostname(), err)
```

Comment:
Let's use [the existing message string variable](https://github.com/pokt-network/pocket/blob/main/libp2p/network/url_conversion.go#L24) here:

```suggestion
		return nil, fmt.Errorf("%s: %s: %w", errResolvePeerIPMsg, peerUrl.Hostname(), err)
```